hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -1015,11 +1016,13 @@ static void validateSecondaryIndexSchemaEvolution(
       }
       
       // Use AvroSchemaCompatibility's field lookup logic to handle aliases
-      Schema.Field writerField = AvroSchemaCompatibility.lookupWriterField(writerSchema, tableField);
+      HoodieSchemaField writerField = HoodieSchemaCompatibility.lookupWriterField(writerSchema, tableField);
       
       if (writerField != null && !tableField.schema().equals(writerField.schema())) {
         // Check if this is just making the field nullable/non-nullable, which is safe from SI perspective
-        if (getNonNullTypeFromUnion(tableField.schema()).equals(getNonNullTypeFromUnion(writerField.schema()))) {
+        HoodieSchema nonNullTableField = HoodieSchemaUtils.getNonNullTypeFromUnion(tableField.schema());
```

Comment:
Let's use the `getNonNullType` method on the `HoodieSchema` when possible

Reply:
Done